types/ember-cli-fastboot/test/assert.ts

```diff
@@ -0,0 +1,5 @@
+/** Static assertion that `value` has type `T` */
+// Disable tslint here b/c the generic is used to let us do a type coercion and
+// validate that coercion works for the type value "passed into" the function.
+// tslint:disable-next-line:no-unnecessary-generics
+export declare function assertType<T>(value: T): void;
```

Comment:
do you falks really need this custom, extra feat, instead of using documented `dtlint` "// $ExpectType" assertions?

Reply:
Historically, we did: it was more robust for checking structural equality than the string name resolution that DTS went provides, which was important for the (extremely complex) legacy Ember class system, which predates ES classes. In this particular library, it is probably less important.

Reply:
OK, but because you're not using modern ESM export list, IMO it will be available for imports, can you put this on the test as internal type? unless you are going to share this within @ember/* ecosystem, which is also OK, but requires clear intent

Reply:
Couple notes here:

- It’s more or less implicitly an internal type already, in that this is not a supported import path for Ember types.
- Unless you *explicitly* do an `import { assertType } from `'ember-cli-fastboot/test/assert';` somewhere in your project, it will *not* be suggested for auto-import.
- This matches how the *rest* of the Ember types do it.
- Despite the rest of the Ember types having done it this way for the last ~5 years, it has never come up as a point of confusion, so I think we’re fine for now.
- Medium-term (within the next ~6 months) we’ll be moving these types *out* of DT and publishing them from Ember itself. (And using [expect-type](https://github.com/mmkal/ts/tree/master/packages/expect-type) for these kinds of assertions!)

Net, I think consistency with the rest of the Ember types is preferable.

Reply:
thansk!